src/client/components/hiluk-menu/hiluk-menu.tsx

```diff
@@ -94,8 +113,24 @@ export class HilukMenu extends React.Component<HilukMenuProps, HilukMenuState> {
     onClose();
   }
 
+  onExportToExternalSystem() {
+    const { essence } = this.props;
+    const viewDefinitionPrinter = new ViewDefinitionPrinter(essence.dataCube.name, DEFAULT_VIEW_DEFINITION_VERSION, defaultDefinitionConverter.toViewDefinition(essence));
+    const viewDefinitionAsJson = viewDefinitionPrinter.printAsJson();
+    const errorHandler = (message: string) => Notifier.failure("Export to external system failed", message);
+
+    exportToExternalSystem(viewDefinitionAsJson, errorHandler)
+      .then(() => {
+        const externalSystem = this.props.externalSystem;
+
+        if (externalSystem && externalSystem.redirectLink) {
+          this.setState({ openExternalSystemView: true });
```

Comment:
Just execute window opening code here. Don't put anything into state that will be used only inside `componentWillUpdate`.